app/pages/sections/catalogue-solutions/routes.js

```diff
@@ -74,16 +75,35 @@ export const catalogueSolutionsRoutes = (authProvider, addContext, sessionManage
 
   router.post('/:orderItemId', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
     const { orderId } = req.params;
+    const accessToken = extractAccessToken({ req, tokenType: 'access' });
+    const solutionName = sessionManager.getFromSession({ req, key: 'solutionName' });
+    const selectedRecipientId = sessionManager.getFromSession({ req, key: 'selectedRecipientId' });
+    const serviceRecipientName = sessionManager.getFromSession({ req, key: 'selectedRecipientName' });
     const selectedPrice = sessionManager.getFromSession({ req, key: 'selectedPrice' });
-
     const response = validateOrderItemForm({ data: req.body, selectedPrice });
+
     if (response.success) {
+      const selectedSolutionId = sessionManager.getFromSession({ req, key: 'selectedSolutionId' });
+      const serviceRecipient = {
+        name: serviceRecipientName,
+        odsCode: selectedRecipientId,
+      };
+      const solution = {
+        id: selectedSolutionId,
+        name: solutionName,
+      };
```

Comment:
I personally would prefer if we didn't do the above 2 in the routes. If you can move this to the controller or even in the formatPostData that would be nice

Reply:
done